pkg/kubectl/cmd/create.go

```diff
@@ -201,8 +201,10 @@ func NameFromCommandArgs(cmd *cobra.Command, args []string) (string, error) {
 
 // CreateSubcommandOptions is an options struct to support create subcommands
 type CreateSubcommandOptions struct {
-	// Name of resource being created
+	// Name of resource instance being created (ex: my-pod, my-namespace)
 	Name string
+	// Resource being created (ex: pod, namespace)
+	Resource string
```

Comment:
Seems like you could simply have the straight `GroupVersionKind` come in here.  It knows what its creating, right?